### PR TITLE
260 submit component

### LINKED ISSUE #260: Submit component

### LINKED ISSUE #265: Dragging over the "upload from your computer" button prevents dropping files in file_upload.vue

### PR DESCRIPTION
**Pull request by james-perretta**
Fixes #260 
Fixes #265

### CODE REVIEW

File: src/components/file_upload.vue [hunk 1]
```diff
@@ -210,6 +210,10 @@ export default class FileUpload extends Vue {
   position: relative;
   text-align: center;
   width: 100%;
+
+  * {
+    pointer-events: none;
```
Comment (ashberg) [inline src/components/file_upload.vue]: The "Upload from your computer" button is no longer clickable.

File: src/components/project_view/submit.vue [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+    }
+  }
+
+  private file_matches_any_patterns(file: File): boolean {
+    for (let expected_file of this.project.expected_student_files) {
+      if (matches(file.name, expected_file.pattern)) {
+        return true;
+      }
+    }
+
```
Comment (ashberg) [inline src/components/project_view/submit.vue]: There's an extra space here.

Comment (james-perretta) [reply src/components/project_view/submit.vue]: Blank line or not before a return statement isn't something that necessarily needs to be standardized, but sure, I can take out this one.

File: src/components/file_upload.vue (hunk 1 above)
Comment (james-perretta) [reply src/components/file_upload.vue]: Oops

**Pull request merged**